### PR TITLE
Support redash v10

### PR DESCRIPTION
**Pull request by minato128**
# Overview

I created a PR based on [the v10 Special instructions](https://github.com/getredash/redash/releases/tag/v10.0.0).

- We used this package and tested it in our environment.
    - https://github.com/minato128/contrib-helm-chart/blob/gh-pages/redash-2.4.1.tgz
        - https://github.com/minato128/contrib-helm-chart/pull/1
        - https://github.com/minato128/contrib-helm-chart/pull/2
- changes
    - image version from v8 to v10
    - add workers
        - `genericWorker`
        - `scheduler`
    - install/upgrade process in `install-upgrade.sh`
    - ingress spec from `networking.k8s.io/v1beta` to `networking.k8s.io/v1`
        - https://github.com/minato128/contrib-helm-chart/pull/3

# References

- https://github.com/getredash/contrib-helm-chart/pull/36
- https://github.com/getredash/contrib-helm-chart/issues/87

### CODE REVIEW

File: templates/configmap.yaml [hunk 1]
```diff
@@ -18,28 +18,43 @@ data:
...
+      fi
+
+      # Upgrade
+      if [[ "${1}" == "upgrade" ]]; then
+        echo "Running Redash database migrations:"
+        timeout $UPGRADE_TIMEOUT /app/manage.py db upgrade
+        echo "Upgrade complete"
+      fi
+
       STATUS=$(timeout $STATUS_TIMEOUT /app/manage.py status 2>&1)
```
[the remark this answers is not in the document]
Comment (grugnog) [reply templates/configmap.yaml:52]: I think switching up the order is fine - were you able to check the 2 cases in the $STATUS check are still appropriate with the status command changes?

**Review by grugnog — COMMENTED** [review]
Wow - this looks great @minato128, thank you so much for working on this!
I think you got all the functional details I can think to check :)

I have one question about the status output. If you have time it would be great to add an upgrade instructions to the Upgrading section README.md.gotmpl.

I also want to run an upgrade test on our internal infra before I merge this - if anyone else is able to run an upgrade test on an their stage environment that would be welcome also.

In terms of releasing I need to add a final 2.x release with a few fixes for the 8.x version. I also want to update the README to drop official support for Helm 2 and some older Kubernetes versions before cutting a 3.0 release. These will only be "policy" changes (with a notice message perhaps) for this 3.0 release but unblock us making some improvements/simplifications down the road without another major release.

**Comment by minato128** [comment]
@grugnog Thank you for your review!

>I have one question about the status output. If you have time it would be great to add an upgrade instructions to the Upgrading section README.md.gotmpl.

I added `From 2.x to 3.x` section. https://github.com/getredash/contrib-helm-chart/pull/102/commits/b634951833f947f8785088c6d91659f0bb03f737

>I also want to run an upgrade test on our internal infra before I merge this - if anyone else is able to run an upgrade test on an their stage environment that would be welcome also.
> In terms of releasing I need to add a final 2.x release with a few fixes for the 8.x version. I also want to update the README to drop official support for Helm 2 and some older Kubernetes versions before cutting a 3.0 release. These will only be "policy" changes (with a notice message perhaps) for this 3.0 release but unblock us making some improvements/simplifications down the road without another major release.

I see. Thanks!

File: templates/configmap.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (minato128) [reply templates/configmap.yaml:52]: Its work in our environment. (upgraded chart from v2.3.1 to this PR version using ArgoCD)
Then I checked the status command logs.


The status command will check redis conn/queue and db status(using latest schema).
https://github.com/getredash/redash/blob/2e67227f1bb6068c3f23bc044ad76ef7d4bccdc7/redash/monitor.py#L53-L59
https://github.com/getredash/redash/blob/2e67227f1bb6068c3f23bc044ad76ef7d4bccdc7/redash/monitor.py#L21-L29

File: values.yaml [hunk 2]
```diff
@@ -355,7 +358,7 @@ ingress:
 adhocWorker:
   # adhocWorker.env -- Redash ad-hoc worker specific envrionment variables.
   env:
-    QUEUES: "queries,celery"
+    QUEUES: "queries"
```
Comment (Jawshua) [inline values.yaml:361]: Hey. I was about to make a PR for the v9 fork we've been running a while, but this one seems to incorporate all of the changes. I wish I'd known about it sooner 😄 

Just wanted to say that we've found that a few extra queues needed to be present in the ad-hoc and scheduled workers (at least in v9):
[PR](https://github.com/getredash/contrib-helm-chart/compare/master...Jawshua:9.0.0#diff-8377b3e3740a3fcd9f682e5fb55425f2fdbece1791854b9e5013e7f1a5e60e7eR361-R399)

Comment (minato128) [reply values.yaml:361]: Thanks 👍 
I used this `docker-compose.yml` as a reference to update workers and queues.

https://github.com/getnewdash/setup/blob/5c739da578f33ab3f9fb088654de2a4de11fcf5d/data/docker-compose.yml

**Comment by yuriipolishchuk** [comment]
We've checked the upgraded helm chart from the PR on our test env - worked like a charm.

@minato128  thank you!

Can we have this merged?

**Comment by minato128** [comment]
@yuriipolishchuk Thanks! 😄 

@grugnog  I tried `git rebase` to fix this conflicts. But I cant fix it. So I recreated #103 PR. 
I understand that some preparations need to be made for the release of Chart v3, but can we merge this PR first?

**Comment by grugnog** [comment]
Closing in favor of #103

**Pull request closed without merging**